### ISSUE TITLE
ETags should only generated on 200 (OK) GET and HEAD

### ISSUE THREAD

**Issue by alexanderkiel** [clojure-liberator/liberator#219]
I have the problem that my `:etag` function is called all the time even on failure handlers like `:handle-malformed`. It makes no sense to have a ETag on a 400 (Bad Request) response.

The relevant section of the current HTTP RFC 7232 can be found [here][1].

[1]: <https://tools.ietf.org/html/rfc7232#section-2.4>

**Comment by alexanderkiel** [comment]
A PUT response with a body and 200 (OK) should probably also carry a ETag.

**Comment by ordnungswidrig** [comment]
That's right: an ETag on a 400 does not make sense. However there's more than 200 OK, for a 304 the ETag (and other headers) are required by [RFC 7232, Section 4.1](https://tools.ietf.org/html/rfc7232#section-4.1).

**Comment by ordnungswidrig** [comment]
I would happily accept a PR if you mind. @alexanderkiel do you have the time to do this?
